Painter/UI/Fleet/FleetViewController.swift

```diff
@@ -144,15 +147,21 @@ extension FleetViewController {
         
         let offset = offsetY - scrollView.contentOffset.y
         
-        if offset <= 0 {
-            UIButton.animate(withDuration: 0.3, delay: 0, options: .curveEaseIn) {
+        if offset <= -1 {
+            UIButton.animate(withDuration: 0.5, delay: 0, options: .curveEaseIn) {
                 self.addButtton.alpha = 0.0
             }
+            UIButton.animate(withDuration: 0.5, delay: 0) { [self] in
+                self.addButtton.transform = CGAffineTransform(scaleX: 0.1, y: 0.1)
+            }
         }
-        else {
-            UIButton.animate(withDuration: 0.3, delay: 0, options: .curveEaseIn) {
+        if offset > 0 {
+            UIButton.animate(withDuration: 0.5, delay: 0, options: .curveEaseIn) {
                 self.addButtton.alpha = 1.0
             }
+            UIButton.animate(withDuration: 0.5, delay: 0) { [self] in
+                self.addButtton.transform = CGAffineTransform(scaleX: 1.0, y: 1.0)
+            }
```

Comment:
extract the duration of floating action button animation somewhere (constant in an enum or something like that)